clippy_lints/src/operators/assign_op_pattern.rs

```diff
@@ -99,3 +113,69 @@ impl<'a, 'tcx> Visitor<'tcx> for ExprVisitor<'a, 'tcx> {
         walk_expr(self, expr);
     }
 }
+
+fn imm_borrows_in_expr(cx: &LateContext<'_>, e: &hir::Expr<'_>) -> hir::HirIdSet {
```

Comment:
thought: should these live in utils?

also worth adding a doc comment

Reply:
I would imagine anything else would need to gather a list of all borrows, not just mutable or immutable ones.

Reply:
ah fair